packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -882,8 +885,9 @@ class _CupertinoModalPopupRoute<T> extends PopupRoute<T> {
 Future<T> showCupertinoModalPopup<T>({
   @required BuildContext context,
   @required WidgetBuilder builder,
+  bool useRootNavigator = true,
 }) {
-  return Navigator.of(context, rootNavigator: true).push(
+  return Navigator.of(context, rootNavigator: useRootNavigator).push(
```

Comment:
Add an assert here and elsewhere that useRootNavigator is not null.

Reply:
Done.